frontend/src/hooks/useSupabaseSession.ts

```diff
@@ -0,0 +1,46 @@
+import { useState, useEffect } from "react";
+import { client } from "../services/supabase";
+import { Session } from "@supabase/supabase-js";
+import { isSessionExpired } from "../utils/timeUtils";
+import { useNavigate } from "react-router-dom";
+import { UserService } from "../services/users/service";
+import { logout } from "../reducer/userSlice";
+import { useDispatch } from "react-redux";
+import { AppDispatch } from "../store/store";
+
+export function useSupabaseSession() {
+  const navigate = useNavigate();
+  const dispatch: AppDispatch = useDispatch();
+  const [session, setSession] = useState<Session | null>(null);
+  const [loading, setLoading] = useState(true);
+
+  useEffect(() => {
+    const getSession = async () => {
+      const result = await client.auth.getSession();
+      const expiredDate = result.data.session?.expires_at;
+      if (!expiredDate) {
```

Comment:
I don't know whether this code works or not. If this works well, you don't need to write an else clause.

```js
if (!expiredDate) {
   console.error("No active session");
  return;
} 
```

Reply:
Okay, I changed it.